### PR TITLE
substitute c++14 for c++17 with more environment variables

### PR DESCRIPTION
**Pull request by mattip**
Maybe this is overkill. At least one of these broke my environment. I don't know which one, but is sure was frustrating to dig around and find this.

### CODE REVIEW

File: working-envs/activate-pytorch-dev.sh [hunk 1]
```diff
@@ -154,6 +154,18 @@ export CFLAGS="$CFLAGS -L$CONDA_PREFIX/lib"
 export CONDA_BUILD_SYSROOT=$CONDA_PREFIX/$HOST/sysroot
 export LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH
 
+if [ -v CXXFLAGS_CONDA_NVCC_BACKUP ] ; then
+    export CXXFLAGS_CONDA_NVCC_BACKUP="`echo $CXXFLAGS_CONDA_NVCC_BACKUP | sed 's/-std=c++17/-std=c++14/'`"
+fi
```
Comment (pearu) [inline working-envs/activate-pytorch-dev.sh:159]: CXXFLAGS_CONDA_NVCC_BACKUP should not be changed, it is needed for deactivating the environment.

File: working-envs/activate-pytorch-dev.sh [hunk 2]
```diff
@@ -154,6 +154,18 @@ export CFLAGS="$CFLAGS -L$CONDA_PREFIX/lib"
...
+if [ -v CXXFLAGS_USED ] ; then
+    export CXXFLAGS_USED="`echo $CXXFLAGS_USED | sed 's/-std=c++17/-std=c++14/'`"
+fi
+if [ -v DEBUG_CXXFLAGS ] ; then
+    export DEBUG_CXXFLAGS="`echo $DEBUG_CXXFLAGS | sed 's/-std=c++17/-std=c++14/'`"
```
Comment (pearu) [inline working-envs/activate-pytorch-dev.sh:164]: that's ok but has a typo in sed command: it misses the tailing `g`

File: working-envs/activate-pytorch-dev.sh [hunk 3]
```diff
@@ -154,6 +154,18 @@ export CFLAGS="$CFLAGS -L$CONDA_PREFIX/lib"
 export CONDA_BUILD_SYSROOT=$CONDA_PREFIX/$HOST/sysroot
 export LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH
 
+if [ -v CXXFLAGS_CONDA_NVCC_BACKUP ] ; then
+    export CXXFLAGS_CONDA_NVCC_BACKUP="`echo $CXXFLAGS_CONDA_NVCC_BACKUP | sed 's/-std=c++17/-std=c++14/'`"
+fi
+if [ -v CXXFLAGS_USED ] ; then
```
Comment (pearu) [inline working-envs/activate-pytorch-dev.sh:160]: what defines CXXFLAGS_USED? Is it used within pytorch build system?

**Review by pearu — CHANGES_REQUESTED** [review]
Thanks for the PR.
But I think that only resetting DEBUG_CXXFLAGS makes sense.

**Comment by mattip** [comment]
I can't really answer any of the questions about these environment variables, I do not know where they came from. Feel free to close this, as it was more of a proof-of-concept of what worked for me.

**Pull request closed without merging**